acestep/text_tasks/external_lm_http_client.py

```
@@ -0,0 +1,54 @@
+    urlopen_impl=None,
+) -> str:
+    """POST JSON to an external provider and return the raw response body."""
+
+    if urlopen_impl is None:
+        urlopen_impl = urllib_request.urlopen
+    req = urllib_request.Request(
+        url=base_url,
+        data=json.dumps(payload).encode("utf-8"),
+        headers=headers,
+        method="POST",
+    )
```

Ruff (0.15.7)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)